### PR TITLE
Introduce `ImmutableEnumSetRules` Refaster rule collection

### PR DESCRIPTION
**Pull request by mohamedsamehsalah**
Suggested commit message
```
Introduce `ImmutableEnumSetRules` refaster rules
```

### CODE REVIEW

File: error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ImmutableEnumSetRules.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+/** Refaster rules related to expressions dealing with immutable {@link Enum} collections. */
+@OnlineDocumentation
+final class ImmutableEnumSetRules {
+  private ImmutableEnumSetRules() {}
+
+  /**
+   * Prefer {@link Sets#immutableEnumSet(Iterable)} for enum collections to take advantage of
+   * internally used {@link EnumSet}.
+   */
+  static final class ImmutableEnumSetIterable<T extends Enum<T>> {
```
Comment (rickie) [inline error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ImmutableEnumSetRules.java]: The yet-to-be formalized Refaster naming algorithm would add the `Sets` here in front :).

**Review by rickie — APPROVED** [review]
Mostly some tweaks to Refaster rule names. Nice new collection @mohamedsamehsalah :rocket:!

Added a commit.

Suggested commit message:
```
Introduce `ImmutableEnumSetRules` Refaster rule collection (#1302)
```

**Review by Stephan202 — APPROVED** [review]
Rebased and added a commit.

**Review by rickie — APPROVED** [review]
Didn't have time to check the changes too thoroughly but they look good!

**Comment by mohamedsamehsalah** [comment]
Thanks @Stephan202 🙏

**Pull request merged**